src/main/org/deidentifier/arx/io/ImportAdapterJDBC.java

```diff
@@ -1,309 +1,419 @@
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.deidentifier.arx.io;
+
+import org.deidentifier.arx.DataType;
+
+import java.io.IOException;
+import java.sql.*;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Import adapter for JDBC
+ * <p>
+ * This adapter can import data from JDBC sources. The source itself is
+ * described by an appropriate {@link ImportConfigurationJDBC} object.
+ *
+ * @author Karol Babioch
+ * @author Fabian Prasser
+ */
+public class ImportAdapterJDBC extends ImportAdapter {
+
+    /**
+     * The configuration describing the CSV file being used.
+     */
+    private ImportConfigurationJDBC config;
+
+    /**
+     * ResultSet containing rows to return.
+     *
+     * @see {@link #next()}
+     */
+    private ResultSet resultSet;
+
+    /**
+     * JDBC statement.
+     */
+    private Statement statement;
+
+    /**
+     * Indicates whether there is another row to return.
+     */
+    private boolean hasNext;
+
+    /**
+     * Indicates whether the first row has already been returned
+     * <p>
+     * The first row contains the name of the columns. Depending upon whether
+     * the name of the column has been assigned explicitly, this is either the
+     * value of the table itself, the value defined by the user.
+     */
+    private boolean headerReturned;
+
+    /**
+     * Number of rows that need to be processed in total.
+     *
+     * @see {@link #getProgress()}
+     */
+    private int totalRows;
+
+    /**
+     * Creates a new instance of this object with given configuration.
+     *
+     * @param config {@link #config}
+     * @throws IOException In case of communication errors with JDBC
+     * @todo Fix IOException
+     */
+    protected ImportAdapterJDBC(ImportConfigurationJDBC config) throws IOException {
+
+        super(config);
+        this.config = config;
+
+        /* Preparation work */
+        indexes = getIndexesToImport();
+        dataTypes = getColumnDatatypes();
+
+        try {
+
+            /* Used to keep track of progress */
+            statement = config.getConnection().createStatement();
+
+            String tableName = config.getTable();
+            String schemaName = config.getSchemaName();
+            String dbName = config.getDbName();
+            DatabaseMetaData meta = config.getConnection().getMetaData();
+            String productName = meta.getDatabaseProductName().toLowerCase();
+
+            String afterTableName = "";
+
+            String databaseType = "";
+
+            if (productName.contains("oracle")) {
```

Comment:
Do you have a reference (e.g. a link) showing that this if statement will work correctly for all types of databases supported? If yes: please add.